### PR TITLE
Invisible touch - this time for real

### PR DESCRIPTION
**Pull request by Majkl-J**
# Document the changes in your pull request

Ports https://github.com/yogstation13/yogstation-2017-/pull/1373 from old yogs.

There are several changes to how this currently works:
Cannot be used on mobs other than humans.
When used on humans, turns the clothes invisible, not the human.
Single use for 16 seconds on 32 second cooldown

# Why is this good for the game?

Adds an alternative to the wall. This actually has a decent gimmick value too, unlike a chair

# Testing

Ran around hitting shit with it like a madman. It did do exactly what intended, and turned a lizard naked

# Spriting
See icondiff bot

# Wiki Documentation
Probably needs to be added for the mime page

# Changelog

:cl:  
rscadd: Added invisible touch to mimes
imageadd: Sprites for invisible touch
/:cl:

### CODE REVIEW

**Comment by MajManatee** [comment]
brain damage clowncil said:
Cant use it on yourself 
Using it on items should make them flicker like it did to the mime in the original PR, not make them invisible.
Shouldnt be usable on weapons, bombs, or anything dangerous like that

**Comment by Maia-J** [comment]
> brain damage clowncil said: Cant use it on yourself Using it on items should make them flicker like it did to the mime in the original PR, not make them invisible. Shouldnt be usable on weapons, bombs, or anything dangerous like that

You literally can't use it on yourself as it has a mob check, the use on weapons means absolutely nothing as they still make a sound and bullets are still visible. If anything maybe you can use it for an ambush, but after the first hit everyone knows what's going on.

Also flicker seems kinda eh, completely ruins the point + would lag the game with inhands updating. Instead of this, I gave the mime only one charge and no flicker.

**Comment by MajManatee** [comment]
Thats what they said dog, its not my choice. That's why i closed it in the first place, because they wouldnt let it have any actual invisibility.

**Comment by Hislyna** [comment]
> brain damage clowncil said: Cant use it on yourself Using it on items should make them flicker like it did to the mime in the original PR, not make them invisible. Shouldnt be usable on weapons, bombs, or anything dangerous like that

Holy shit thats some cringe

**Comment by Hislyna** [comment]
impassable wall and box all gucci, make something invis for 16 seconds and HOLY SHIT FAM. Jesus christ we need to stop making this shit all about balance its cringe

**Comment by Maia-J** [comment]
> impassable wall and box all gucci, make something invis for 16 seconds and HOLY SHIT FAM. Jesus christ we need to stop making this shit all about balance its cringe

Turns out that isn't even a real decision it's just manatee overreacting because he can see balance council channels

**Comment by iamthesilly** [comment]
Would be funnier if it was a full minute instead of 16 seconds, might have time to actually pull more interesting pranks with it

**Comment by LaOubliette** [comment]
> Would be funnier if it was a full minute instead of 16 seconds, might have time to actually pull more interesting pranks with it

Im gonna cry :’)

**Comment by thegoldencat413** [comment]
uh manatee is reading the wrong thread or something this PR hasn’t been talked about by balance clowncil at all yet

**Comment by MajManatee** [comment]
It was, then it was closed, and a new one wasn't re-opened

**Review by cuackles — APPROVED** [review]

File: code/modules/spells/spell_types/touch/invisible_touch.dm [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	drop_message = span_notice("You let the power from your hand dissipate.")
+
+	var/list/things = list()
+	var/list/blacklist = list (
+        /obj/machinery/nuclearbomb/selfdestruct,
+        /obj/structure/closet
+        )
+
+/datum/action/cooldown/spell/touch/invisible_touch/is_valid_target(atom/cast_on)
+	return TRUE
```
Comment (Moltijoe) [inline code/modules/spells/spell_types/touch/invisible_touch.dm:38]:
```suggestion
	return (!isturf(cast_on))
```
this way it won't get wasted on turf

**Review by Moltijoe — CHANGES_REQUESTED** [review]

File: code/modules/spells/spell_types/touch/invisible_touch.dm (hunk 1 above)
Comment (Maia-J) [reply code/modules/spells/spell_types/touch/invisible_touch.dm:38]: Interesting thing: This was tried. I do not know why but it didn't work so I moved the check elsewhere

Comment (Maia-J) [reply code/modules/spells/spell_types/touch/invisible_touch.dm:38]: See line 37-38

Comment (Moltijoe) [reply code/modules/spells/spell_types/touch/invisible_touch.dm:38]: Its because touch is currently bugged, i have a pr fixing it

Comment (Maia-J) [reply code/modules/spells/spell_types/touch/invisible_touch.dm:38]: You are fixing a completely different proc that doesn't even call this

Comment (Moltijoe) [reply code/modules/spells/spell_types/touch/invisible_touch.dm:38]: i'm fixing a proc that calls a proc that calls this

Comment (Maia-J) [reply code/modules/spells/spell_types/touch/invisible_touch.dm:38]: Well tell me after it is merged, not 3 days before it

Comment (Moltijoe) [inline code/modules/spells/spell_types/touch/invisible_touch.dm:38]:
```suggestion
	return (!isturf(cast_on))
```

update your branch first and then do this

**Review by Moltijoe — CHANGES_REQUESTED** [review]

**Review by Moltijoe — APPROVED** [review]

**Pull request merged**